### PR TITLE
Add rubocop gem and config

### PR DESCRIPTION
**Pull request by egglestn**
Rubocop gem and config only, with no fixes - the fixes seemed pretty high risk, so this just enables developers to run and fix for themselves.

At time of commit, currently rubocop was finding 338 offenses

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - Rakefile
+  - '**/Gemfile'
+  - config.ru
+  Exclude:
+  - config/**/*
+  - db/**/*
+  - spec/**/*
+  - bin/**/*
+  - lib/tasks/*
+  TargetRubyVersion: 2.3
```
Comment (baob) [inline .rubocop.yml]: Maybe update this ... it may reduce the number of autocorrects.

Comment (egglestn) [reply .rubocop.yml]: Fixed, but sadly no, still 338 offenses. We could increase the maxes a bit more, and catch a few more of the method length issues, for example

**Comment by baob** [comment]
I'm in favour ... defaults are not all what I'd choose, but I think consistency is more important than those choices, so all good from me.

**Comment by aspiers** [comment]
Great, thanks!

**Pull request merged**